### PR TITLE
PAR-421 additional features for enforcement notifications

### PR DESCRIPTION
**Pull request by jon-edery**
This PR adds custom access control to ensure the correct users can/cannot access versions of the notification actions dictated by custom business logic.

Custom flow link added to enforcement notifications view to control the displaying of actions that need to be reviewed or acted on depending on the logged in user role (enforcement officer / primary authority user).

Review controller added to display the resulting acted upon enforcement notification.

### CODE REVIEW

**Comment by stemount** [comment]
This looks like good work to me. :)

File: web/modules/features/par_enforcement_flows/src/Controller/ParEnforcementFlowsCompletedEnforcementController.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+      }
+    }
+    return parent::accessCallback();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function titleCallback() {
+    return;
```
Comment (stemount) [inline web/modules/features/par_enforcement_flows/src/Controller/ParEnforcementFlowsCompletedEnforcementController.php]: don't fancy a  title? heh

Comment (jon-edery) [reply web/modules/features/par_enforcement_flows/src/Controller/ParEnforcementFlowsCompletedEnforcementController.php]: Its no needed  now but Im sure the designs will change when we start getting client feedback... this is for future updates.

File: sync/par_data.par_data_enforcement_action_type.enforcement_action.yml [hunk 2]
```diff
@@ -36,3 +36,4 @@ configuration:
       allowed: Approved
       blocked: Blocked
       referred: Referred
+      approved: Approved
```
Comment (kalpaitch) [inline sync/par_data.par_data_enforcement_action_type.enforcement_action.yml]: Don't want to change the statuses.

File: web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php [hunk 3]
```diff
@@ -67,6 +67,7 @@ class ParDataEnforcementAction extends ParDataEntity {
   const APPROVED = 'approved';
```
Comment (kalpaitch) [inline web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php]: Yea this should be changed to 'allowed'

Comment (kalpaitch) [reply web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php]: or 'approved' and change the original migrated values in a hook update

File: web/modules/features/par_data_config/config/install/par_data.par_data_enforcement_action_type.enforcement_action.yml [hunk 4]
```diff
@@ -33,3 +33,4 @@ configuration:
       allowed: Approved
       blocked: Blocked
       referred: Referred
+      approved: Approved
```
Comment (kalpaitch) [inline web/modules/features/par_data_config/config/install/par_data.par_data_enforcement_action_type.enforcement_action.yml]: allowed is approved. let's remove.

File: web/modules/features/par_enforcement_flows/par_enforcement_flows.routing.yml [hunk 5]
```diff
@@ -123,6 +123,7 @@ par_enforcement_flows.approve:
     _title: 'Approve Enforcement Notice'
   requirements:
     _permission: 'approve enforcement notice'
+    _custom_access: '\Drupal\par_enforcement_flows\Form\ParEnforcementApproveNoticeForm::accessCallback'
```
Comment (kalpaitch) [inline web/modules/features/par_enforcement_flows/par_enforcement_flows.routing.yml]: Nice.

File: web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php (hunk 3 above)
Comment (jon-edery) [reply web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php]: Ill set this to "allowed" to resolve this issue...

Comment (jon-edery) [reply web/modules/custom/par_data/src/Entity/ParDataEnforcementAction.php]: will check the db to confirm no updates are needed.

**Pull request merged**